src/vulkan/render.zig

```diff
@@ -192,48 +192,29 @@ const dispatch_override_functions = std.StaticStringMap(CommandDispatchType).ini
     .{ "vkCreateInstance", .base },
 });
 
-// Functions that return an array of objects via a count and data pointer.
-const enumerate_functions = std.StaticStringMap(void).initComptime(.{
-    .{"vkEnumeratePhysicalDevices"},
-    .{"vkEnumeratePhysicalDeviceGroups"},
-    .{"vkGetPhysicalDeviceQueueFamilyProperties"},
-    .{"vkGetPhysicalDeviceQueueFamilyProperties2"},
-    .{"vkEnumerateInstanceLayerProperties"},
-    .{"vkEnumerateInstanceExtensionProperties"},
-    .{"vkEnumerateDeviceLayerProperties"},
-    .{"vkEnumerateDeviceExtensionProperties"},
-    .{"vkGetImageSparseMemoryRequirements"},
-    .{"vkGetImageSparseMemoryRequirements2"},
-    .{"vkGetDeviceImageSparseMemoryRequirements"},
-    .{"vkGetPhysicalDeviceSparseImageFormatProperties"},
-    .{"vkGetPhysicalDeviceSparseImageFormatProperties2"},
-    .{"vkGetPhysicalDeviceToolProperties"},
-    .{"vkGetPipelineCacheData"},
-
-    .{"vkGetPhysicalDeviceSurfaceFormatsKHR"},
-    .{"vkGetPhysicalDeviceSurfaceFormats2KHR"},
-    .{"vkGetPhysicalDeviceSurfacePresentModesKHR"},
-
-    .{"vkGetSwapchainImagesKHR"},
-    .{"vkGetPhysicalDevicePresentRectanglesKHR"},
-
-    .{"vkGetPhysicalDeviceCalibrateableTimeDomainsKHR"},
-});
-
-// Given one of the above commands, returns the type of the array elements
-// (and performs some basic verification that the command has the expected signature).
-fn getEnumerateFunctionDataType(command: reg.Command) !reg.TypeInfo {
+// If the given function returns an array (e.g. `vkEnumeratePhysicalDevices()`),
+// returns the type of the array elements.
+fn getEnumerateFunctionDataType(name: []const u8, command: reg.Command) ?reg.TypeInfo {
     if (command.params.len < 2) {
-        return error.InvalidRegistry;
+        return null;
     }
     const count_param = command.params[command.params.len - 2];
-    if (!count_param.is_buffer_len) {
-        return error.InvalidRegistry;
-    }
     const data_param = command.params[command.params.len - 1];
+    if (!(count_param.is_buffer_len and data_param.is_optional)) {
+        return null;
+    }
+
+    // Uses a bool out-parameter for an "incomplete" result, and so would need a
+    // special case implementation.
+    if (mem.eql(u8, name, "vkGetFaultData")) {
+        return null;
+    }
+
+    // TODO: Extra validation.
+
```

Comment:
Just some thoughts:
- Did you consider checking the mutability of the output parameter? That should be something like `param_type.pointer_is_mutable`, or simply use `classifyParam` and check `out_pointer` (I think). Additionally, you could check that the size param has class `mut_buffer_len`
- Did you consider checking whether the successcodes includes `VK_INCOMPLETE`?

Maybe this would be enough and then we don't need to bother with the scripts and stuff